vendor-extra/JatsContentBundle/tests/EventListener/BuildView/FrontPdfContentHeaderListenerTest.php

```diff
@@ -0,0 +1,213 @@
+    {
+        $listener = new FrontPdfContentHeaderListener();
+
+        $element = $this->loadElement(
+            <<<XML
+<?xml version="1.0" encoding="UTF-8"?>
+<front xmlns="http://jats.nlm.nih.gov" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <article-meta>
+        <self-uri content-type="pdf" xlink:href="http://example.com"/>
+    </article-meta>
+</front>
+XML
+        );
+
+        $event = new BuildViewEvent(
+            $element,
+            new TemplateView('@LiberoPatterns/content-header.html.twig', ['download' => 'bar'])
+        );
+        $listener->onBuildView($event);
+        $view = $event->getView();
+
+        $this->assertInstanceOf(TemplateView::class, $view);
+        $this->assertSame('@LiberoPatterns/content-header.html.twig', $view->getTemplate());
+        $this->assertSame(['download' => 'bar'], $view->getArguments());
+        $this->assertEmpty($view->getContext());
+    }
+
+    /**
+     * @test
+     * @dataProvider downloadProvider
+     */
+    public function it_sets_the_download_argument(string $xml, array $expected) : void
+    {
+        $listener = new FrontPdfContentHeaderListener();
+
+        $element = $this->loadElement($xml);
+
+        $context = ['bar' => 'baz'];
+
+        $event = new BuildViewEvent(
+            $element,
+            new TemplateView('@LiberoPatterns/content-header.html.twig', [], $context)
+        );
+        $listener->onBuildView($event);
+        $view = $event->getView();
+
+        $this->assertInstanceOf(TemplateView::class, $view);
+        $this->assertSame('@LiberoPatterns/content-header.html.twig', $view->getTemplate());
+        $this->assertEquals(['download' => $expected], $view->getArguments());
+        $this->assertSame(['bar' => 'baz'], $view->getContext());
+    }
+
+    public function downloadProvider() : iterable
+    {
+        yield 'absolute' => [
+            <<<XML
+<?xml version="1.0" encoding="UTF-8"?>
+<front xmlns="http://jats.nlm.nih.gov" xmlns:xlink="http://www.w3.org/1999/xlink">
+    <article-meta>
+        <self-uri content-type="pdf" xlink:href="http://example.com/file.pdf"/>
+    </article-meta>
+</front>
+XML
+            ,
+            [
+                'link' => [
+                    'href' => 'http://example.com/file.pdf',
+                ],
+            ],
+        ];
+
+        yield 'xml:base' => [
+            <<<XML
+<?xml version="1.0" encoding="UTF-8"?>
+<front xmlns="http://jats.nlm.nih.gov" xmlns:xlink="http://www.w3.org/1999/xlink" xml:base="http://example.com/">
+    <article-meta>
+        <self-uri content-type="pdf" xlink:href="file.pdf"/>
+    </article-meta>
+</front>
+XML
+            ,
+            [
+                'link' => [
+                    'href' => 'http://example.com/file.pdf',
+                ],
+            ],
+        ];
+    }
+
+    /**
+     * @test
+     */
+    public function it_does_nothing_if_the_uri_is_relative() : void
```

Comment:
is this meant to be replaced later? Or just not supported?

Reply:
Not supported. No where to load it from.